cmd/nerdctl/create_linux_test.go

```diff
@@ -106,3 +108,31 @@ func TestCreateWithMACAddress(t *testing.T) {
 		}
 	}
 }
+
+func TestCreateWithTty(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("json-file log driver is not yet implemented on Windows")
+	}
+	base := testutil.NewBase(t)
+	imageName := testutil.CommonImage
+	withoutTerminalContainerName := "without-terminal-" + testutil.Identifier(t)
+	withTerminalContainerName := "with-terminal-" + testutil.Identifier(t)
+
+	// without -t
+	base.Cmd("create", "--name", withoutTerminalContainerName, imageName).AssertOK()
+	// start fail
+	base.Cmd("start", withoutTerminalContainerName).AssertOK()
+	defer base.Cmd("container", "rm", "-f", withoutTerminalContainerName).AssertOK()
+	withoutTerminalContainer := base.InspectContainer(withoutTerminalContainerName)
+	assert.Equal(base.T, "exited", withoutTerminalContainer.State.Status)
+	assert.Equal(base.T, false, withoutTerminalContainer.State.Running)
+
+	// with -t
+	base.Cmd("create", "-t", "--name", withTerminalContainerName, imageName).AssertOK()
+	// start success
+	base.Cmd("start", withTerminalContainerName).AssertOK()
+	defer base.Cmd("container", "rm", "-f", withTerminalContainerName).AssertOK()
+	withTerminalContainer := base.InspectContainer(withTerminalContainerName)
+	assert.Equal(base.T, "running", withTerminalContainer.State.Status)
+	assert.Equal(base.T, true, withTerminalContainer.State.Running)
```

Comment:
The test has to verify that tty is allocated when `-t` is specified.

Probably you can just test `stty`

```console
$ docker run -d --name without-t alpine stty
f6ee65e70c512ddba7846f595f3dc5ce0cdb72110e9f901d1661e05f0a8a1372
$ docker logs without-t 
stty: standard input: Not a tty
$ docker container inspect without-t  | jq .[0].State.ExitCode
1
```

```console
$ docker run -td --name with-t alpine stty
5b7bf5c0b3b07af1ccbe2c049aa34841ea635e10c0b493e056201e16dd38de03
$ docker logs with-t 
speed 38400 baud; line = 0;
intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = <undef>; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R; werase = ^W; lnext = ^V; flush = ^O; min = 1; time = 0;
-brkint -imaxbel
$ docker container inspect with-t  | jq .[0].State.ExitCode
0
```

Reply:
> The test has to verify that tty is allocated when `-t` is specified.
> 
> Probably you can just test `stty`
> 
> ```
> $ docker run -d --name without-t alpine stty
> f6ee65e70c512ddba7846f595f3dc5ce0cdb72110e9f901d1661e05f0a8a1372
> $ docker logs without-t 
> stty: standard input: Not a tty
> $ docker container inspect without-t  | jq .[0].State.ExitCode
> 1
> ```
> 
> ```
> $ docker run -td --name with-t alpine stty
> 5b7bf5c0b3b07af1ccbe2c049aa34841ea635e10c0b493e056201e16dd38de03
> $ docker logs with-t 
> speed 38400 baud; line = 0;
> intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = <undef>; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R; werase = ^W; lnext = ^V; flush = ^O; min = 1; time = 0;
> -brkint -imaxbel
> $ docker container inspect with-t  | jq .[0].State.ExitCode
> 0
> ```

done